src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -312,6 +335,146 @@ static RachisConsensus()
             {
                 StartIndex = 0
             });
+
+            LogHistoryTable = new TableSchema();
+            LogHistoryTable.DefineKey(new TableSchema.SchemaIndexDef
+            {
+                StartIndex = (int)LogHistoryColumn.Guid,
+            });
+
+            LogHistoryTable.DefineFixedSizeIndex(new TableSchema.FixedSizeSchemaIndexDef
+            {
+                Name = LogHistoryIndexSlice,
+                StartIndex = (int)LogHistoryColumn.Index
+            });
+        }
+
+        public unsafe void InsertHistoryLog(TransactionOperationContext context, long index, long term, BlittableJsonReaderObject cmd)
+        {
+            var guid = GetGuidFromCommand(cmd);
+            if (guid == null) // shouldn't happened in new cluster version!
+                return;
+
+            if (guid == string.Empty)
+                return;
+
+            var table = context.Transaction.InnerTransaction.OpenTable(LogHistoryTable, LogHistorySlice);
+            var type = GetTypeFromCommand(cmd);
```

Comment:
why do you care about the type?

Reply:
mainly for debug... if we will ever wonder what the hell this command was about, at least we will have the type.